Jenkinsfile.promote

```diff
@@ -470,3 +514,80 @@ void runMavenDeploy(MavenCommand mvnCmd) {
     }
     mvnCmd.skipTests(true).run('clean deploy')
 }
+
+def updateAndResetBranch(String repo, String branchToReset){
+    githubscm.createBranch(branchToReset)
+
+    removeJbossNexusFromMavenAndGradle()
+
+    githubscm.commitChanges("[${getBuildBranch()}] Update ${branchToReset} to ${getProjectVersion()}", {
+        githubscm.findAndStageNotIgnoredFiles('pom.xml')
+        githubscm.findAndStageNotIgnoredFiles('build.gradle')
+    })
+    forcePushProtectedBranch(repo, branchToReset)
+}
+
+void removeJbossNexusFromMavenAndGradle() {
+        sh '''
+            sed -i \':a;N;$!ba;s/\\n *<repositories>.*<\\/repositories>//g\' */pom.xml
+            sed -i \':a;N;$!ba;s/\\n *<repositories>.*<\\/repositories>//g\' pom.xml
+            sed -i \':a;N;$!ba;s/\\n *\\/\\/ To get snapshots during development[^{}]*maven {[^{}]*mavenContent {[^{}]*snapshotsOnly[^{}]*}[^{}]*}//g\' */build.gradle
+        '''
+        assert !sh (script: 'grep -Rn "repository.jboss.org" --include={pom.xml,build.gradle}|cat'
+                , allowEmptyResults: true, returnStdout: true)
+}
+
+String getGhCredsID(){
+    return params.GITHUB_TOKEN_CREDS_ID;
+}
+
+String getProtectionMapScript(){
+    return "if .restrictions == null then . + {\"restrictions_used\":null}  else . + {\"restrictions_used\":{\"users\":[.restrictions.users[].login], \"team\":[.restrictions.users[].slug]}} end | {\"required_status_checks\": .required_status_checks,  \"required_pull_request_reviews\": {\"dismiss_stale_reviews\":(.required_pull_request_reviews.dismiss_stale_reviews//false), \"require_code_owner_reviews\":(.required_pull_request_reviews.require_code_owner_reviews//false)}, \"enforce_admins\":.enforce_admins.enabled, \"restrictions\":.restrictions_used}";
```

Comment:
Is it possible to make this line more readable (starting by breaking it into multiple lines)?

Reply:
Let me just put down the concern we discussed: what if the JSON format changes? We don't want to silently lose any settings.

Reply:
Added in 6ee6bd3 Thanks for the idea!